Aplib.Core/Metadata.cs

```diff
@@ -0,0 +1,37 @@
+using System;
+
+namespace Aplib.Core
+{
+    /// <summary>
+    /// Data structure to store information about a component which may be useful for debugging or logging.
+    /// </summary>
+    public class Metadata
+    {
+        /// <summary>
+        ///     Gets the unique identifier of the component.
+        /// </summary>
+        public Guid Id { get; }
+        
+        /// <summary>
+        /// Gets the name used to display the component during debugging, logging, or general overviews.
+        /// </summary>
+        public string? Name { get; }
+
+        /// <summary>
+        /// Gets the description used to describe the component during debugging, logging, or general overviews.
+        /// </summary>
+        public string? Description { get; }
+
+        /// <summary>
+        /// Store information about a component which may be useful for debugging or logging or general overviews.
+        /// </summary>
+        /// <param name="name">The name used to display the component.</param>
+        /// <param name="description">The description used to describe the component.</param>
+        public Metadata(string? name = null, string? description = null)
+        {
+            Id = Guid.NewGuid();
+            Name = name;
+            Description = description;
+        }
```

Comment:
Could you add an (internal) constructor which also takes an ID. This will be better when testing with metadata.